examples/module/dynamics/tank_robot.py

```diff
@@ -0,0 +1,115 @@
+import math
+import pypose as pp
+import torch as torch
+import matplotlib.pyplot as plt
+from matplotlib.patches import Ellipse
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+class TankRobot(pp.module.System):
+    def __init__(self, dt, Q, R):
+        super(TankRobot, self).__init__()
+        self.state = torch.tensor([0.0, 0.0, 0.0])
+        self.input = torch.tensor([0.0, 0.0])
+        self.dt = dt
+        self.register_buffer("Q", torch.eye(self.A.shape) if Q is None else Q)
+        self.register_buffer("R", torch.eye(self.C.shape[0]) if R is None else R)
+        self.set_refpoint()
+
+    def state_transition(self, state, input, t=None):
+        # input is torch.tensor([v, w]) where v is forward velo and w is angular velo
+        input_dt = torch.mul(input, self.dt)
+        theta = state[2] + input_dt[1]
+        v_x = input_dt[0] * torch.cos(theta)
```

Comment:
I think here is the delta of the distances of x and y. Maybe the name of the d_x (d_y) is better